src/Symfony/Component/Security/Guard/Firewall/GuardAuthenticationListener.php

```diff
@@ -172,12 +190,12 @@ public function setRememberMeServices(RememberMeServicesInterface $rememberMeSer
      * Checks to see if remember me is supported in the authenticator and
      * on the firewall. If it is, the RememberMeServicesInterface is notified.
      *
-     * @param GuardAuthenticatorInterface $guardAuthenticator
-     * @param Request                     $request
-     * @param TokenInterface              $token
-     * @param Response                    $response
+     * @param AuthenticatorInterface $guardAuthenticator
+     * @param Request                $request
+     * @param TokenInterface         $token
+     * @param Response               $response
      */
-    private function triggerRememberMe(GuardAuthenticatorInterface $guardAuthenticator, Request $request, TokenInterface $token, Response $response = null)
+    private function triggerRememberMe($guardAuthenticator, Request $request, TokenInterface $token, Response $response = null)
```

Comment:
IMO we should be consistent in removing the type hint (like here) or not (like in `GuardAuthenticatorHandler` below). Keeping the type hint might be a good idea to catch all places where we need to adjust them when removing code in 4.0.

Reply:
👍  I re-added the typehint in this class so that it breaks if we forget to change them when removing the deprecated interface.